rpc/gnmi/gnmi-commit-confirmed.md

```diff
@@ -0,0 +1,106 @@
+defined in the RFC with the intention that most of the gRPC API clients are going to
+be automated systems and the proto should facilitate a simpler implementation of
+client workflows and server implementation.
+
+The server can be viewed as a singleton resource, at any given point in time there
+can be only one commit active. This commit can be either confirmed or canceled
+before a new commit can begin. Client is expected to provide full configuration
+during the commit request, the commit cannot be amended once issued.
+
+# 3. Definition
+
+A `Commit` message is embedded the Extension message of the SetRequest proto.
+
+## 3.1 Proto
+
+```
+// Commit confirmed extension allows automated revert of the configuration after
+// certain duration if an explicit confirmation is not issued. It allows explicit
+// cancellation of the commit during the rollback window. There cannot be more
+// than one commit active at a given time.
+// The document about gNMI commit confirmed can be found at
+// https://github.com/openconfig/reference/blob/master/rpc/gnmi/gnmi-commit-confirmed.md
+message Commit {
+  // ID is provided by the client during the commit request. During confirm and cancel
+  // actions the provided ID should match the ID provided during commit.
+  // If ID is not passed in any actions server shall return error.
+  // Required.
+  string id = 1;
+  oneof action {
+    // commit action creates a new commit. If a commit is on-going, server returns error.
+    CommitRequest commit = 2;
+    // confirm action will confirm an on-going commit, the ID provided during confirm
+    // should match the on-going commit ID.
+    CommitConfirm confirm = 3;
+    // cancel action will cancel an on-going commit, the ID provided during cancel
+    // should match the on-going commit ID.
+    CommitCancel cancel = 4;
+  }
+}
+
+// CommitRequest is used to create a new confirmed commit. It hold additional
+// parameter requried for commit action.
+message CommitRequest {
+  // Maximum duration to wait for a confirmaton before reverting the commit.
+  google.protobuf.Duration rollback_duration = 1;
+}
+
+// CommitConfirm is used to confirm an on-going commit. It hold additional
+// parameter requried for confirm action.
+message CommitConfirm {}
+
+// CommitCancel is used to cancel an on-going commit. It hold additional
+// parameter requried for cancel action.
+message CommitCancel {}
+```
+
+## 3.2 SetRequest handling                                                        
+
+### 3.2.1 Commit
+A commit can be initiated by providing `CommitRequest` as action in the extension. A `id` must to be
+provided by the client. The server shall associate the commit with the provided `id`.
+During confirm or cancel action the provided `id` must match the `id` of the on-going commit.
+`rollback_duration` can be used to override the default rollback duration which is 10min.
```

Comment:
shall we explicitly call out that the server should maintain a default 10m interval when rollback_duration is not present?

Reply:
additionally, it might be worth repeating in what units the rollback is provided. It is not easy to find the googleapis repo where the typedef is set.

Reply:
Updated.